converter/grpc_interceptor.go

```diff
@@ -55,1360 +45,78 @@ type PayloadCodecGRPCClientInterceptorOptions struct {
+		Outbound: &proxy.VisitPayloadsOptions{
+			Visitor: func(vpc *proxy.VisitPayloadsContext, payloads []*commonpb.Payload) ([]*commonpb.Payload, error) {
+				var err error
+				for i := len(options.Codecs) - 1; i >= 0; i-- {
+					if payloads, err = options.Codecs[i].Encode(payloads); err != nil {
+						return payloads, err
+					}
+				}
 
-type serviceInterceptor struct {
-	codecs []PayloadCodec
-}
+				return payloads, nil
+			},
+			SkipSearchAttributes: true,
+		},
+		Inbound: &proxy.VisitPayloadsOptions{
+			Visitor: func(vpc *proxy.VisitPayloadsContext, payloads []*commonpb.Payload) ([]*commonpb.Payload, error) {
+				var err error
+				for _, codec := range options.Codecs {
+					if payloads, err = codec.Decode(payloads); err != nil {
+						return payloads, err
+					}
+				}
 
-func (s *serviceInterceptor) encodePayload(payload *commonpb.Payload) error {
-	var err error
-	payloads := []*commonpb.Payload{payload}
-	for i := len(s.codecs) - 1; i >= 0; i-- {
-		if payloads, err = s.codecs[i].Encode(payloads); err != nil {
-			return err
-		}
-	}
-	if len(payloads) != 1 {
-		return fmt.Errorf("received %d payloads from codec, expected 1", len(payloads))
-	}
-	*payload = *payloads[0]
-	return nil
+				return payloads, nil
+			},
+			SkipSearchAttributes: true,
+		},
+	})
 }
 
-func (s *serviceInterceptor) encodePayloads(payloadspb *commonpb.Payloads) error {
-	var err error
-	payloads := payloadspb.Payloads
-	for i := len(s.codecs) - 1; i >= 0; i-- {
-		if payloads, err = s.codecs[i].Encode(payloads); err != nil {
-			return err
-		}
-	}
-	payloadspb.Payloads = payloads
-	return nil
+type NewFailureGRPCClientInterceptorOptions struct {
+	// DataConverter ??
+	EncodeCommonAttributes bool
 }
 
-func (s *serviceInterceptor) decodePayload(payload *commonpb.Payload) error {
-	var err error
-	payloads := []*commonpb.Payload{payload}
-	for _, codec := range s.codecs {
-		if payloads, err = codec.Decode(payloads); err != nil {
-			return err
-		}
-	}
-	if len(payloads) != 1 {
-		return fmt.Errorf("received %d payloads from codec, expected 1", len(payloads))
-	}
-	*payload = *payloads[0]
-	return nil
+// Do we want to expose this from converter so that we are sure things line up?
+type encodedFailure struct {
+	Message    string `json:"message"`
+	StackTrace string `json:"stack_trace"`
 }
 
-func (s *serviceInterceptor) decodePayloads(payloadspb *commonpb.Payloads) error {
-	var err error
-	payloads := payloadspb.Payloads
-	for _, codec := range s.codecs {
-		if payloads, err = codec.Decode(payloads); err != nil {
-			return err
-		}
+// NewFailureGRPCClientInterceptor returns a GRPC Client Interceptor that will mimic the encoding
+// that the SDK system would perform when configured with a FailureConverter with the EncodeCommonAttributes option set.
+func NewFailureGRPCClientInterceptor(options NewFailureGRPCClientInterceptorOptions) (grpc.UnaryClientInterceptor, error) {
```

Comment:
May want to add doc that says "If using with NewPayloadCodecGRPCClientInterceptor, make sure this interceptor comes first in the chain"